src/MoBi.Presentation/Tasks/Interaction/BuildingBlockTaskRetriever.cs

```diff
@@ -45,62 +45,64 @@ public IInteractionTasksForBuildingBlock TaskFor(IBuildingBlock buildingBlock)
          }
       }
 
-      private void retrieveTask<TBuildingBlock>(TBuildingBlock buildingBlock) where TBuildingBlock: class, IBuildingBlock
+      private void retrieveTask<TParent, TBuildingBlock>(TBuildingBlock buildingBlock) where TBuildingBlock : class, IBuildingBlock where TParent : class
       {
-         _task = _container.Resolve<IInteractionTasksForBuildingBlock<TBuildingBlock>>();
+         _task = _container.Resolve<IInteractionTasksForBuildingBlock<TParent, TBuildingBlock>>();
       }
-      private void retrieveTask<TBuildingBlock>() where TBuildingBlock : class, IBuildingBlock
+
+      private void retrieveTask<TParent, TBuildingBlock>() where TBuildingBlock : class, IBuildingBlock where TParent : class
       {
-         _task = _container.Resolve<IInteractionTasksForBuildingBlock<TBuildingBlock>>();
+         _task = _container.Resolve<IInteractionTasksForBuildingBlock<TParent, TBuildingBlock>>();
       }
- 
+
       public void Visit(MoleculeBuildingBlock objToVisit)
       {
-         retrieveTask(objToVisit);
+         retrieveTask<Module, MoleculeBuildingBlock>(objToVisit);
```

Comment:
why do you specifrcy Module all the time here? Cannot the line 55 use Module instead?